x-pack/plugins/global_search/common/types.ts

```diff
@@ -87,3 +87,28 @@ export interface GlobalSearchBatchedResults {
    */
   results: GlobalSearchResult[];
 }
+
+/**
+ * Search parameters for the {@link GlobalSearchPluginStart.find | `find` API}
+ *
+ * @public
+ */
+export interface GlobalSearchFindParams {
+  /**
+   * The term to search for. Can be undefined if searching by filters.
+   */
+  term?: string;
+  /**
+   * The types of results to search for.
+   */
+  types?: string[];
+  /**
+   * The tag ids to filter search by.
+   */
+  tags?: string[];
+}
+
+/**
+ * @public
+ */
+export type GlobalSearchProviderFindParams = GlobalSearchFindParams;
```

Comment:
It may, depending on how we decide to internally process the parameters form the service before forwarding them to providers. Also atm the service and provider types are all separated, so it was mostly for consistency.